packages/-ember-data/tests/integration/model-errors-test.ts

```diff
@@ -67,4 +70,24 @@ module('integration/model.errors', function (hooks) {
 
     assert.dom('.error-list__error').doesNotExist();
   });
+
+  test('Uncommitted model can become valid after changing 2 erred fields', async function (this: CurrentTestContext, assert) {
+    this.tag = this.owner.lookup('service:store').createRecord('tag');
```

Comment:
> we should assert the state name and state flags generated for each part of the transition.

:+1:

> Likely this test should call save ... as manually adding errors is not a well supported or encouraged pattern.

it's a public API and it's the exact case when this issue happens. Also, in the test above we don't save anything.

Reply:
when you try to save an object with relations (like when using `EmbeddedRecordsMixin`) ember doesn't know how to assign the errors so you have to do it manually.

Reply:
Even though it's a trick and might not be as great semantically, I think many people did it one way or another (especially since `lid` is not documented anywhere and didn't exist when many apps were created), so why not fix this behaviour. It won't do any harm imo.

Reply:
@yratanov I'm not saying don't make it work :P Just want to ensure we test what we actually support. I guess it makes sense for these edge cases, I respect folks have historically done bad things around transactional saves. Can we get test coverage of both?

Reply:
`currentState` is a private property, can you advise on how to test it here?

On `save`, the record will become `invalid`, therefore it will have `becomeValid` substate. The error appears only for `uncommitted` records.

Reply:
currentState on models isn't private per-say. Though I think we do want to deprecate at least part of what it does (the state name calculation) in favor of folks using the flags only.

I'm going to skip on this for this release cycle as I believe this is probably fixed on master given we've removed the state machine. If it's not lets make tweaks, if it is lets use this as the basis for a fix for older releases and target them.

Reply:
@runspired it does work in master! Can we use the fix from this branch for 3.28? We can't migrate straight to 4.X right now...

Reply:
Feel free to retarget the PR. Would be great to still get a good test for this onto master though.

Reply:
Opened new pr against `lts-3-28`, I hope it's the correct one? :)